lib/smart_proxy_dns_powerdns/backend/mysql.rb

```diff
@@ -35,15 +35,20 @@ def create_record domain_id, name, type, content
       name = connection.escape(name)
       content = connection.escape(content)
       type = connection.escape(type)
-      connection.query("INSERT INTO records (domain_id, name, ttl, content, type) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}')")
+      connection.query("INSERT INTO records (domain_id, name, ttl, content, type, change_date) VALUES (#{domain_id}, '#{name}', #{ttl.to_i}, '#{content}', '#{type}', UNIX_TIMESTAMP())")
       connection.affected_rows == 1
     end
 
     def delete_record domain_id, name, type
       name = connection.escape(name)
       type = connection.escape(type)
       connection.query("DELETE FROM records WHERE domain_id=#{domain_id} AND name='#{name}' AND type='#{type}'")
-      connection.affected_rows >= 1
+      ret = connection.affected_rows >= 1
+      if ret
+        connection.query("UPDATE records SET change_date=UNIX_TIMESTAMP() WHERE domain_id=#{domain_id} AND type='SOA'")
+        ret = connection.affected_rows == 1
```

Comment:
> You are right, logging would definitely help! The thing is I am not a Rubyist and don't know much about the log interface of the SmartProxy. Any advice would be appreciated.

I'm not a Rubyist either so I'm doing the best I can, but the plugin logs very little. Currently there is only one example in this code base: https://github.com/theforeman/smart_proxy_dns_powerdns/blob/566cdec3c8d5335ee0cbecf6bdeb4c7a6488756d/lib/smart_proxy_dns_powerdns/backend/rest.rb#L87 and I'm not sure the extends is needed there since the main class also does that so first try without it.

> Also, I think the README needs to be updated to guide to the SOA-EDIT properties defined in PowerDNS. I honestly totally missed this part of the PDNS guide at first and this is a required setting in order to have a working, auto-incremented SOA serial.

Absolutely.

> Regarding your last usecase I don't think this is true: update will change the SOA record even if the current change_date is already set. My guess is that the backend will still return affected_rows == 1 is this case because nothing in the SQL statement filters the SOA records with "up to date" change_date field.

This is not true for MySQL. An ```UPDATE``` statement will set ```affected_rows``` to the number of rows that changed. If the column already contains the value then it will not increase the ```affected_rows```. Since unix timestamps have a resolution of 1 second there's a decent chance this will happen.